### PR TITLE
Authentication plugin

### PR DESCRIPTION
**Pull request by pierre-vigier**
Related to #131 , First POC and draft of an implementation strategy

### CODE REVIEW

**Comment by clintongormley** [comment]
@pierre-vigier sorry for the delay. I haven't forgotten, just have a lot on my plate at the moment.

**Comment by pierre-vigier** [comment]
No worries, just let me know when you have time

**Comment by pjlsergeant** [comment]
Huge +1 for this

File: lib/Search/Elasticsearch/CxnAuth/Basic.pm [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            path     => $uri->path,
+            userinfo => $uri->userinfo
+        };
+    }
+
+    my $userinfo = $node->{userinfo} || $params->{userinfo} || '';
+
+    if( $userinfo ) {
+        require MIME::Base64;
+        my $auth = MIME::Base64::encode_base64($userinfo);
```
Comment (warewolf) [inline lib/Search/Elasticsearch/CxnAuth/Basic.pm]: Bug here, fixed in https://github.com/elastic/elasticsearch-perl/pull/160.  Feel free to lift the fix.

**Comment by pjlsergeant** [comment]
Thanks. Returned the favour and added the test that was wanted

**Comment by warewolf** [comment]
:+1: thanks @pjlsergeant, much appreciated.  I was praying someone who had better familiarity with how Elastic does tests would be able to throw a test together.

**Pull request closed without merging**